### ISSUE TITLE
Error with creating user database DB_NAME

### ISSUE THREAD

**Issue by skandyla** [Khezen/docker-mongo-rocks#7]
Hi! 
First, thanks for the image. It seems to be pretty good.
However there is some error with creating the user database.

When specify DB_NAME variable, scripts try to use it for authentication the admin.
i.e. 
`mongo -u admin -p adminpass --authenticationDatabase someuserdb --eval db.createUser({user: 'someuser', pwd: 'somepass', roles:[{role: 'dbOwner', db: 'someuserdb'}]});`

I suppose the solution could be to specify the default authenticationDatabase for admin to `admin`. Or to have some variable like ADMIN_AUTHENTICATION_DATABASE.  

### How to reproduce the error:
Create docker-compose.yml:  
```
version: '2'
services:
    mongod1:
        image: khezen/mongo:3.4
        environment:
            AUTH: y
            ADMIN_USER: admin
            ADMIN_PWD: adminpass
            DB_NAME:   someuserdb
            DB_USER: someuser
            DB_PWD:  somepass
        volumes:
             - dbdata:/data/db
        ports:
             - "27017:27017"
        network_mode: bridge
        restart: always

volumes:
    dbdata:
```

Start docker-compose:  
`$ docker-compose up`:

### Example output:
```
Creating volume "dockercomposemongodb_dbdata" with default driver
Creating dockercomposemongodb_mongod1_1
Attaching to dockercomposemongodb_mongod1_1
mongod1_1  | /run/mongod/entrypoint.sh
mongod1_1  | => Waiting for confirmation of MongoDB service startup
mongod1_1  | mongod --storageEngine wiredTiger --port 27017 --dbpath /data/db --oplogSize 50 --keyFile /config/key
mongod1_1  | 2017-02-17T17:19:48.359+0000 I STORAGE  [main] Compression: snappy
mongod1_1  | 2017-02-17T17:19:48.359+0000 I STORAGE  [main] MaxWriteMBPerSec: 1024
mongod1_1  | 2017-02-17T17:19:48.359+0000 I STORAGE  [main] Crash safe counters: 0
mongod1_1  | 2017-02-17T17:19:48.359+0000 I STORAGE  [main] Counters: 1
mongod1_1  | 2017-02-17T17:19:48.359+0000 I STORAGE  [main] Use SingleDelete in index: 0
mongod1_1  | 2017-02-17T17:19:48.372+0000 I CONTROL  [initandlisten] MongoDB starting : pid=17 port=27017 dbpath=/data/db 64-bit host=3f3a86cd5d2f
mongod1_1  | 2017-02-17T17:19:48.372+0000 I CONTROL  [initandlisten] db version v3.4.2
mongod1_1  | 2017-02-17T17:19:48.372+0000 I CONTROL  [initandlisten] git version: 3f76e40c105fc223b3e5aac3e20dcd026b83b38b
mongod1_1  | 2017-02-17T17:19:48.372+0000 I CONTROL  [initandlisten] allocator: tcmalloc
mongod1_1  | 2017-02-17T17:19:48.372+0000 I CONTROL  [initandlisten] modules: rocks
mongod1_1  | 2017-02-17T17:19:48.372+0000 I CONTROL  [initandlisten] build environment:
mongod1_1  | 2017-02-17T17:19:48.372+0000 I CONTROL  [initandlisten]     distarch: x86_64
mongod1_1  | 2017-02-17T17:19:48.372+0000 I CONTROL  [initandlisten]     target_arch: x86_64
mongod1_1  | 2017-02-17T17:19:48.372+0000 I CONTROL  [initandlisten] options: { net: { port: 27017 }, replication: { oplogSizeMB: 50 }, security: { keyFile: "/config/key" }, storage: { dbPath: "/data/db", engine: "wiredTiger" } }
mongod1_1  | 2017-02-17T17:19:48.376+0000 I STORAGE  [initandlisten]
mongod1_1  | 2017-02-17T17:19:48.376+0000 I STORAGE  [initandlisten] ** WARNING: Using the XFS filesystem is strongly recommended with the WiredTiger storage engine
mongod1_1  | 2017-02-17T17:19:48.376+0000 I STORAGE  [initandlisten] **          See http://dochub.mongodb.org/core/prodnotes-filesystem
mongod1_1  | 2017-02-17T17:19:48.376+0000 I STORAGE  [initandlisten] wiredtiger_open config: create,cache_size=487M,session_max=20000,eviction=(threads_max=4),config_base=false,statistics=(fast),log=(enabled=true,archive=true,path=journal,compressor=snappy),file_manager=(close_idle_time=100000),checkpoint=(wait=60,log_size=2GB),statistics_log=(wait=0),
mongod1_1  | 2017-02-17T17:19:48.874+0000 I CONTROL  [initandlisten] ** WARNING: You are running this process as the root user, which is not recommended.
mongod1_1  | 2017-02-17T17:19:48.874+0000 I CONTROL  [initandlisten]
mongod1_1  | 2017-02-17T17:19:49.013+0000 I FTDC     [initandlisten] Initializing full-time diagnostic data capture with directory '/data/db/diagnostic.data'
mongod1_1  | 2017-02-17T17:19:49.261+0000 I INDEX    [initandlisten] build index on: admin.system.version properties: { v: 2, key: { version: 1 }, name: "incompatible_with_version_32", ns: "admin.system.version" }
mongod1_1  | 2017-02-17T17:19:49.261+0000 I INDEX    [initandlisten] 	 building index using bulk method; build may temporarily use up to 500 megabytes of RAM
mongod1_1  | 2017-02-17T17:19:49.268+0000 I INDEX    [initandlisten] build index done.  scanned 0 total records. 0 secs
mongod1_1  | 2017-02-17T17:19:49.269+0000 I COMMAND  [initandlisten] setting featureCompatibilityVersion to 3.4
mongod1_1  | 2017-02-17T17:19:49.270+0000 I NETWORK  [thread1] waiting for connections on port 27017
mongod1_1  | 2017-02-17T17:19:53.381+0000 I NETWORK  [thread1] connection accepted from 127.0.0.1:40548 #1 (1 connection now open)
mongod1_1  | 2017-02-17T17:19:53.381+0000 I ACCESS   [conn1] note: no users configured in admin.system.users, allowing localhost access
mongod1_1  | 2017-02-17T17:19:53.382+0000 I NETWORK  [conn1] received client metadata from 127.0.0.1:40548 conn1: { application: { name: "MongoDB Shell" }, driver: { name: "MongoDB Internal Client", version: "3.4.2" }, os: { type: "Linux", name: "PRETTY_NAME="Debian GNU/Linux stretch/sid"", architecture: "x86_64", version: "Kernel 4.9.8-moby" } }
mongod1_1  | 2017-02-17T17:19:53.387+0000 I -        [conn1] end connection 127.0.0.1:40548 (1 connection now open)
mongod1_1  | mongo admin --eval db.createUser({user: 'admin', pwd: 'adminpass', roles:[{role: 'root', db: 'admin'}]});
mongod1_1  | MongoDB shell version v3.4.2
mongod1_1  | connecting to: mongodb://127.0.0.1:27017/admin
mongod1_1  | 2017-02-17T17:19:53.428+0000 I NETWORK  [thread1] connection accepted from 127.0.0.1:40550 #2 (1 connection now open)
mongod1_1  | 2017-02-17T17:19:53.429+0000 I NETWORK  [conn2] received client metadata from 127.0.0.1:40550 conn2: { application: { name: "MongoDB Shell" }, driver: { name: "MongoDB Internal Client", version: "3.4.2" }, os: { type: "Linux", name: "PRETTY_NAME="Debian GNU/Linux stretch/sid"", architecture: "x86_64", version: "Kernel 4.9.8-moby" } }
mongod1_1  | MongoDB server version: 3.4.2
mongod1_1  | 2017-02-17T17:19:53.604+0000 I COMMAND  [conn2] command admin.system.users appName: "MongoDB Shell" command: insert { insert: "system.users", documents: [ { _id: "admin.admin", user: "admin", db: "admin", credentials: { SCRAM-SHA-1: { iterationCount: 10000, salt: "bkOzqKC+nvVQdMdFL5kNmQ==", storedKey: "Ul12cN9iDIc8AdIhM4QWH0IwBrc=", serverKey: "dC5xzvDQN7Tq8409y9cH8CV2XZo=" } }, roles: [ { role: "root", db: "admin" } ] } ] } ninserted:1 keysInserted:1 numYields:0 reslen:44 locks:{ Global: { acquireCount: { r: 10, w: 4 } }, Database: { acquireCount: { r: 3, W: 4 } }, Collection: { acquireCount: { r: 3, w: 3 } } } protocol:op_query 160ms
mongod1_1  | 2017-02-17T17:19:53.604+0000 I COMMAND  [conn2] command admin.system.version appName: "MongoDB Shell" command: createUser { createUser: "admin", pwd: "xxx", roles: [ { role: "root", db: "admin" } ], digestPassword: false, writeConcern: { w: "majority", wtimeout: 300000.0 } } numYields:0 reslen:22 locks:{ Global: { acquireCount: { r: 10, w: 4 } }, Database: { acquireCount: { r: 3, W: 4 } }, Collection: { acquireCount: { r: 3, w: 3 } } } protocol:op_command 173ms
mongod1_1  | Successfully added user: {
mongod1_1  | 	"user" : "admin",
mongod1_1  | 	"roles" : [
mongod1_1  | 		{
mongod1_1  | 			"role" : "root",
mongod1_1  | 			"db" : "admin"
mongod1_1  | 		}
mongod1_1  | 	]
mongod1_1  | }
mongod1_1  | 2017-02-17T17:19:53.610+0000 I -        [conn2] end connection 127.0.0.1:40550 (1 connection now open)
mongod1_1  | mongo -u admin -p adminpass --authenticationDatabase someuserdb --eval db
mongod1_1  | MongoDB shell version v3.4.2
mongod1_1  | connecting to: mongodb://127.0.0.1:27017
mongod1_1  | 2017-02-17T17:19:53.652+0000 I NETWORK  [thread1] connection accepted from 127.0.0.1:40552 #3 (1 connection now open)
mongod1_1  | 2017-02-17T17:19:53.653+0000 I NETWORK  [conn3] received client metadata from 127.0.0.1:40552 conn3: { application: { name: "MongoDB Shell" }, driver: { name: "MongoDB Internal Client", version: "3.4.2" }, os: { type: "Linux", name: "PRETTY_NAME="Debian GNU/Linux stretch/sid"", architecture: "x86_64", version: "Kernel 4.9.8-moby" } }
mongod1_1  | MongoDB server version: 3.4.2
mongod1_1  | 2017-02-17T17:19:53.655+0000 I ACCESS   [conn3] SCRAM-SHA-1 authentication failed for admin on someuserdb from client 127.0.0.1:40552 ; UserNotFound: Could not find user admin@someuserdb
mongod1_1  | 2017-02-17T17:19:53.656+0000 E QUERY    [thread1] Error: Authentication failed. :
mongod1_1  | DB.prototype._authOrThrow@src/mongo/shell/db.js:1459:20
mongod1_1  | @(auth):6:1
mongod1_1  | @(auth):1:2
mongod1_1  | 2017-02-17T17:19:53.660+0000 I -        [conn3] end connection 127.0.0.1:40552 (1 connection now open)
mongod1_1  | exception: login failed
mongod1_1  | mongo -u admin -p adminpass --authenticationDatabase someuserdb --eval db.createUser({user: 'someuser', pwd: 'somepass', roles:[{role: 'dbOwner', db: 'someuserdb'}]});
mongod1_1  | MongoDB shell version v3.4.2
mongod1_1  | connecting to: mongodb://127.0.0.1:27017
mongod1_1  | 2017-02-17T17:19:53.700+0000 I NETWORK  [thread1] connection accepted from 127.0.0.1:40554 #4 (1 connection now open)
mongod1_1  | 2017-02-17T17:19:53.701+0000 I NETWORK  [conn4] received client metadata from 127.0.0.1:40554 conn4: { application: { name: "MongoDB Shell" }, driver: { name: "MongoDB Internal Client", version: "3.4.2" }, os: { type: "Linux", name: "PRETTY_NAME="Debian GNU/Linux stretch/sid"", architecture: "x86_64", version: "Kernel 4.9.8-moby" } }
mongod1_1  | MongoDB server version: 3.4.2
mongod1_1  | 2017-02-17T17:19:53.703+0000 I ACCESS   [conn4] SCRAM-SHA-1 authentication failed for admin on someuserdb from client 127.0.0.1:40554 ; UserNotFound: Could not find user admin@someuserdb
mongod1_1  | 2017-02-17T17:19:53.704+0000 E QUERY    [thread1] Error: Authentication failed. :
mongod1_1  | DB.prototype._authOrThrow@src/mongo/shell/db.js:1459:20
mongod1_1  | @(auth):6:1
mongod1_1  | @(auth):1:2
mongod1_1  | 2017-02-17T17:19:53.705+0000 I -        [conn4] end connection 127.0.0.1:40554 (1 connection now open)
mongod1_1  | exception: login failed
mongod1_1  | /run/mongod/start.sh
```

**Comment by khezen** [comment]
Thank for reporting. Patch will be released soon.

**Comment by khezen** [comment]
@skandyla 
Patch has been released for 3.4.2 & 3.4.2-slim.
